designs/helm-chart-autogenerate-plugin.md

```diff
@@ -0,0 +1,271 @@
+    └── values.yaml
+```
+
+To reduce the user adaptation, the command
+leverage the existing kustomize default directory and create the similar layout of the helm chart from it. The generated `values.yaml` is as follows:
+
+```yaml
+# global configurations
+nameOverride: ""
+
+fullnameOverride: ""
+
+## CRD configuration under the `config/crd` directory
+crd:
+  ## Whether the `resources` field contains `crd` in the `config/default/kustomization.yaml` file
+  create: false
+
+## RBAC configuration under the `config/rbac` directory
+rbac:
+  ## Whether the `resources` field contains `rbac` in the `config/default/kustomization.yaml` file
+  create: true
+  serviceAccountName: "controller-manager"
+
+
+## Manager configuration referent to `config/manager/manager.yaml`
+manager:
+  ## The Security Context of the manager Pod.
+  ## More info: https://kubernetes.io/docs/tasks/configure-pod-container/security-context/#set-the-security-context-for-a-pod
+  podSecurityContext:
+    runAsNonRoot: true
+
+  ## Configure the kubeRbacProxy container.
+  kubeRbacProxy:
+    image:
+     registry: gcr.io
+     repository: "kubebuilder/kube-rbac-proxy"
+     tag: v0.13.0
+     pullPolicy: IfNotPresent
+
+  ## Configure the resources.
+  ## More info: https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/ 
+  resources:
+    limits:
+      cpu: 500m
+      memory: 128Mi
+    requests:
+      cpu: 5m
+      memory: 64Mi
+
+  ## Configure the controller container.
+  controller:
+    image:
+      registry: docker.io
+      repository: ""
+      tag: latest
+      pullPolicy: IfNotPresent
+
+     ## The environment variables of the controller container.
+     ## More info: https://kubernetes.io/docs/tasks/inject-data-application/define-environment-variable-container/
+  	env: []
+
+    ## Configure extra options for liveness probe.
+    ## More info: https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-probes/#configure-probes
+    livenessProbe:
+      enabled: true
+      httpGet:
+        path: /healthz
+        port: 8081
+      initialDelaySeconds: 15
+      periodSeconds: 20
+
+    readinessProbe:
+      enabled: true
+      httpGet:
+        path: /readyz
+        port: 8081
+      initialDelaySeconds: 5
+      periodSeconds: 10
+
+    ## Configure the resources.
+    ## More info: https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/ 
+    resources:
+      limits:
+        cpu: 500m
+        memory: 128Mi
+      requests:
+        cpu: 10m
+        memory: 64Mi
+
+     ## Configure the scheduling rules.
+     ## More info: https://kubernetes.io/docs/concepts/configuration/assign-pod-node/#affinity-and-anti-affinity
+     affinity: {}
+
+     ## More info: https://kubernetes.io/docs/concepts/configuration/assign-pod-node/#nodeselector
+     nodeSelector: {}
+
+     ## More info: https://kubernetes.io/docs/concepts/configuration/taint-and-toleration
+     tolerations: {}
```

Comment:
Same here. The values must be exactly only what we have in the default scaffold
We need to stick with each otherwise will be very hard we maintain and make it work